### DISCUSSION TITLE
a turborepo pipeline task that depends on a root package.json task that does not execute in each package

### DISCUSSION THREAD

**Discussion by dagda1** [vercel/turborepo#1247] [Help] [answered]
I have a root package.json task that does not execute in every package that calls `yarn tsc`.

Currently, I have it as a `prebuild` script in the root package.json:

```json
"prebuild": "yarn tsc",
```
Is there anyway to specify this as a `dependsOn` element of `build` in the `turborepo.json`?

If I add `tsc` like below then it will get executed in the each package which is not what I want.

```json
"pipeline": {
  "tsc": {
    "dependsOn": ["generate"],
    "outputs": ["dist-types/**"]
  },
  "build": {
    "dependsOn": ["tsc", "^build"],
    "outputs": ["dist/**"]
  },
```

**Comment by minkimcello** [comment]
Are you trying to _prevent_ packages from running `yarn tsc` even though it _can_? Or are you trying to avoid running `yarn tsc` for packages that do not _have_ the script?

If it's the latter, you don't have to worry. Unlike `yarn workspaces run x`, turborepo won't error out if a script isn't available for certain packages.

if you're trying to prevent certain packages from running the script even though it could, you can use the `--filter` flag:
```
turbo run build --filter=!@foo/bar
```

**Reply by dagda1** [reply]
I want it to run but I cannot use it in a `dependsOn` array because it is a root level script.  I found [this PR](https://github.com/vercel/turborepo/pull/1040#pullrequestreview-939899052) that is not merged yet but hopefully will.

**Reply by minkimcello** [reply]
Oh right. I imagine a temporary workaround for the time being could be to specify a root script like this:
```
"tscBeforeBuild": "yarn prebuild && yarn turbo run build"
```

**Reply by dagda1** [reply]
@minkimcello that might work, thanks.

**Comment by anthonyshew** [comment, marked as answer]
You can depend on a root task using `//#prebuild`
```
    "dependsOn": ["//#prebuild"]
```

**Reply by alextbok** [reply]
Thanks! I found this fairly deep down in the documentation so linking to where I found it for posterity: https://turbo.build/messages/missing-root-task-in-turbo-json#solution
